### DISCUSSION TITLE
Disappearing soft keyboard in Android app

### DISCUSSION THREAD

**Discussion by PedroRF123154** [apache/cordova#498] [General]
Soft keyboard disappearing in Android app

I have this problem with many users, I found a page about this and I tried several published functions but the problem persists.

![w6X1l](https://github.com/user-attachments/assets/a06ddd24-5c89-4d91-9264-9677b2a0e4ed)

https://stackoverflow.com/questions/38296574/disappearing-soft-keyboard-in-android-app

Any solution for this?

**Comment by breautek** [comment]
With the information available I can't really say for certain what the issue is but I can share knowledge that can potentially help pinpoint the root issue.

Based on the animated gif, it definitely looks like the input is losing focus, just after the window is resized. On Android the document "window" is resized when the soft keyboard opens matching the new size of the view port. This is part of `adjustResize` behaviour.

## adjustResize vs adjustPan

`adjustPan` is another configuration which causes the native view to pan instead of resizing. This means the view will not change in height to make room for the keyboard, and instead the entire view will pan. The view is not scrollable with this mode, so it can hide content (whereas `adjustResize` gives developers the flexibility of creating scrollable views). If `adjustPan` does solve the issue, it just simply highlights that the application is resetting focus when it shouldn't, either on window resize or orientation change.

These are native configurations, which can should be editable via [&lt;edit-config&gt;](https://cordova.apache.org/docs/en/11.x/plugin_ref/spec.html#edit-config) directive. I wouldn't call using `adjustPan` a solution, but depending on your use case it could be something that is "good enough".

## Node replacement

If you're swapping nodes which some frameworks like (ReactJS) might do depending on how the application is doing it's rendering, it might cause the behaviour you're seeing. If you're using any UI frameworks like ReactJS, or Angular, etc... review their documentation to ensure that they manipulate existing nodes rather than a complete replacement. Replacing nodes will cause nodes to lose their focus.

**Reply by PedroRF123154** [reply]
With adjustPan the problem persists. The only thing I think it could be is the phonegap-plugin-mobile-accessibility plugin.

window.MobileAccessibility.usePreferredTextZoom(false);

I'll try to uninstall it and let you know.

Thanks

**Reply by PedroRF123154** [reply]
The problem has not been solved yet. I am looking and trying to clean up. The problem I have is that it happens to users and I do not have access to the mobile to detect what is causing it. Thanks

**Reply by PedroRF123154** [reply]
The problem was found on Samsung and Xiaomi devices. On Samsung, after updating to ONE UI 6.1, this problem occurs and they recommend resetting or reinstalling the keyboard and, as a last option, resetting the phone to factory mode. The same for Xiaomi.

Thanks
